File: leetcode_1_two-sum/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            // key: number, value: index
            map.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            int result = target - nums[i];
            if (map.containsKey(result) && map.get(result) != i) {
                return new int[] { i, map.get(result) };
            }
        }
        return null;
    }
}

File: leetcode_1_two-sum/Solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = [];

    for (let i = 0; i < nums.length; i++) {
        // key: number, value: index
        map[nums[i]] = i;
    }

    for (let i = 0; i < nums.length; i++) {
        const result = target - nums[i];

        if (result in map) {
            if (map[result] !== i) {
                return [map[result], i];
            }
        }
    }

    return null;
};
